packages/core/src/compile.ts

```diff
@@ -150,9 +149,10 @@ function serialiseTranslationRuntime(
 
     if (Object.keys(params).length > 0) {
       const formatGeneric = hasTags ? '<T = string>' : '';
-      const formatReturn = hasTags
-        ? 'string | T | Array<string | T>'
-        : 'string';
```

Comment:
I know it's a small change, but can we move this to a separate PR (and merge it before releasing this PR)? IMO moving types to the core package feels big enough that it should be a standalone change.